### PR TITLE
Projector Locking strategies

### PR DESCRIPTION
**Pull request by boekkooi-fresh**
Due to some performance issues I started working on adding a Skip Lock based projector but this spun a little out of control.
In the PR:
- the Aggragate and Stream Projector where moved to the sql package (from postgres)
- the concept of a projector transaction was introduced
- Added a useFieldLocked to the advisory projector. This allows the locked update field to not be used.

### CODE REVIEW

File: driver/sql/projector_notification.go [hunk 1]
```diff
@@ -135,25 +132,28 @@ func (s *NotificationProjector) project(
...
-	notification *driverSQL.ProjectionNotification,
-	rawState *driverSQL.ProjectionRawState,
-	stream goengine.EventStream,
+	tx ProjectorTransaction,
+	stream *eventStreamHandlerIterator,
 ) error {
 	var (
-		state         driverSQL.ProjectionState
-		stateAcquired bool
+		err   error
```
Comment (sergiughf) [inline driver/sql/projector_notification.go]: Maybe we can move this to the return of the function? So ultimately we use 3 less lines here.

```
func (s *notificationProjector) projectStream(...) (err error) {
...
}
```

File: driver/sql/sql.go [hunk 2]
```diff
@@ -16,3 +18,45 @@ type (
...
 )
+
+// AcquireConn will return a new connection
+func AcquireConn(ctx context.Context, db *sql.DB) (*sql.Conn, error) {
+	var (
+		timeoutCtx    context.Context
+		timeoutCancel context.CancelFunc
+	)
+
+	for i := 0; i < 3; i++ {
```
Comment (sergiughf) [inline driver/sql/sql.go]: Why do we have 3 here and not let's say 5 or something else?

Comment (boekkooi-fresh) [reply driver/sql/sql.go]: This is similar to https://github.com/golang/go/blob/master/src/database/sql/sql.go#L1390

**Comment by sergiughf** [comment]
+1 LGTM! :)

**Pull request merged**